Documentation/Evolution/RegexSyntax.md

```diff
@@ -0,0 +1,245 @@
+
+#### Alternation
+
+```
+Regex       -> '' | Alternation
+Alternation -> Concatenation ('|' Concatenation)*
+```
+
+This is the operator with the lowest precedence in a regular expression, and checks if any of its branches match the input.
+
+#### Concatenation
+
+```
+Concatenation   -> (!'|' !')' ConcatComponent)*
+ConcatComponent -> Trivia | Quote | Quantification
+```
+
+Implicitly denoted by adjacent expressions, a concatenation matches against a sequence of regular expression patterns. This has a higher precedence than an alternation, so e.g `abc|def` matches against `abc` or `def`. The `ConcatComponent` token varies across engine, but at least matches some form of trivia, e.g comments, quoted sequences e.g `\Q...\E`, and a quantified expression.
+
+#### Quantification
+
+```
+Quantification -> QuantOperand Quantifier?
+Quantifier     -> ('*' | '+' | '?' | '{' Range '}') QuantKind?
+QuantKind      -> '?' | '+'
+```
+
+Specifies that the operand may be matched against a certain number of times.
+
+#### Groups
+
+```
+GroupStart    -> '(?' GroupKind | '('
+GroupKind     -> ':' | '|' | '>' | '=' | '!' | '*' | '<=' | '<!' | '<*'
+               | NamedGroup | MatchingOptionSeq (':' | ')')
+            
+NamedGroup    -> 'P<' GroupNameBody '>'
+               | '<' GroupNameBody '>'
+               | "'" GroupNameBody "'"
+
+GroupNameBody -> Identifier
+```
+
+Groups define a new scope within which a recursive regular expression pattern may occur. Groups have different semantics depending on how they are introduced, some may capture the nested match, some may match against the input without advancing, some may change the matching options set in the new scope, etc.
+
+#### Anchors
+
+```
+Anchor -> '^' | '$' | '\b'
+```
+
+Anchors match against a certain position in the input rather than on a particular character of the input.
+
+#### Unicode scalars
+
+
+
+#### Builtin character classes
+
+
+
+#### Custom character classes
+
+```
+CustomCharClass -> Start Set (SetOp Set)* ']'
+Start           -> '[' '^'?
+Set             -> Member+
+Member          -> CustomCharClass | !']' !SetOp (Range | Atom)
+Range           -> Atom `-` Atom
+```
+
+Custom characters classes introduce their own language, in which most regular expression metacharacters become literal
+
+
+#### Character properties
+
+### PCRE-specific syntax
+
+#### Callouts
+
+### Oniguruma-specific syntax
+
+#### Custom reference syntax
+
+#### Callout syntax
+
+#### Absent functions
+
+### ICU-specific syntax
+
+
+
+### .NET-specific syntax
+
+#### Balancing groups
+
+```
+GroupNameBody -> Identifier | Identifier? '-' Identifier
+```
+
+.NET supports the ability for a group to reference a prior group, causing the prior group to be deleted, and any intermediate matched input to become the capture of the current group.
+
+#### Character class subtraction with `-`
+
+
+
+## Syntactic differences between engines
+
+### Conflicting differences
```

Comment:
Maybe just an intro paragraph and then everything listed is a potential conflict with our demonstrated resolution strategy. I.e. everything gets elevated one level of markdown-nesting .